sregistry/main/s3/__init__.py

```diff
@@ -79,18 +81,29 @@ def get_bucket(self):
 
         self.bucket = self.s3.Bucket(self.bucket_name)
         # check if the bucket exists and can be accessed
+        # based on https://stackoverflow.com/a/47565719/1253230
         try:
             self.s3.meta.client.head_bucket(Bucket=self.bucket.name)
-        except ClientError:
+        except ClientError as e:
             self.bucket = None
- 
-        # If the bucket doesn't exist, try creating it
-        if self.bucket is None:
-            try:
-                self.bucket = self.s3.create_bucket(Bucket=self.bucket_name)
-                bot.info('Created bucket %s' % self.bucket.name )
-            except ClientError as e:
-                bot.exit('Could not create bucket {}: {}'.format(self.bucket_name, str(e)))
+            error_code = int(e.response['Error']['Code'])
+            if error_code == 403:
+                # print("Private Bucket. Forbidden Access!")
+                if not self._id or not self._key:
+                    bot.exit('The {} bucket is not public and needs AWS credentials to be accessed'.format(self.bucket_name))
+                bot.exit('The {} bucket cannot be accessed with the provided AWS credentials'.format(self.bucket_name))
+            elif error_code == 404:
```

Comment:
So are we sure that if the bucket *does* exist but it's private (and owned by someone else) that it will never return 404? (Some web APIs do this).

Reply:
Yep, tested this locally, by creating a bucket with my root credentials, and then trying to access it with AWS credentials that allow me to access just a single, different bucket.

Reply:
ok, great!